cmd/network-console-collector/config.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"encoding/json"
+	"os"
+	"time"
+)
+
+type Config struct {
+	APIListenAddress     string
+	APIDisableAccessLogs bool
+	TLSCert              string
+	TLSKey               string
+
+	EnableConsole   bool
+	ConsoleLocation string
+	PrometheusAPI   string
```

Comment:
Doubt: The collector api works as a reverse proxy for the prometheus v1 http api so that the console ui can run queries. If we aspire to support more than our very rudimentary unsecured network-console-prometheus instance this may become complicated. Think TLS config, different auth methods, etc.

Reply:
I agree that we can do better than Auth basic, but it may not be worth putting in too much effort. 
However, the reason we use a minimum level of protection is not for the console but to protect the APIs, and not just those of the collector, which honestly don't seem to contain sensitive data, but those of Prometheus. In the past, we had an issue with QE because we were publicly exposing the Prometheus route, which means exposing sensitive data if someone do a query. So, I think we will face the same problem if we don't protect the APIs from the collector in a situation without OCP auth proxy.

Reply:
Instead if we want to think about a solution maybe we keep in stanby keykloack but it would be another bomb to maintain and manage. Keycloak is currently used in Skupper X by Ted to grant API ACLs based on potential user permissions